### PR TITLE
Add/reader/refresh refactor content safety rules

### PR DESCRIPTION
**Pull request by samouri**
In this PR, i've taken the three rules: `safeContentImages`, `makeEmbedsSecure`, and `detectEmbeds` renamed/modified them.

Before the PR they did the following:
1.` makeImagesSafe`: made images 'safe' and also created post.content_images
2. `makeEmbedsSafe`: made embeds safer by sandboxing them.  Also removed any bad iframes.
3. `detectEmbeds`: create post.content_embeds and remove the sandbox in some cases.

I believe I've created a more logical split in this PR between cleaning/detecting media:
1.`makeImagesSafe`: just make them safe
2. `makeEmbedsSafe`: just make embeds safe.  removed redundant sandbox logic.
3. `detectMedia`: scans post to create content_embeds, content_images, and also content_media.  content_media contains an ordered list of usable media (a pre-req for #8555).

Still todo for this to land:
- [ ]  thoroughly test
- [x] fix unit tests to raise confidence. I don't think this PR can get shorter because its moving around a bunch of files / the unit tests need to happen at the same time

### CODE REVIEW

File: client/lib/post-normalizer/rule-content-safe-images.js [hunk 1]
```diff
@@ -1,160 +1,86 @@
...
-				if ( newSrcSet ) {
-					image.setAttribute( 'srcset', newSrcSet );
-				} else {
-					image.removeAttribute( 'srcset' );
-				}
-			}
+	// trickery to remove it from the dom / not load the image
+	// TODO: test if this is necessary
+	if ( ! safeSource || imageShouldBeRemovedFromContent( imgSource ) ) {
+		image.remove();
```
Comment (samouri) [inline client/lib/post-normalizer/rule-content-safe-images.js]: need to change this to parent.removeChild(..)

File: client/lib/post-normalizer/rule-content-safe-images.js [hunk 2]
```diff
@@ -1,160 +1,86 @@
...
-				} ).filter( imgSrc => imgSrc.url );
-				const newSrcSet = srcset.stringify( imgSrcSet );
-				if ( newSrcSet ) {
-					image.setAttribute( 'srcset', newSrcSet );
-				} else {
-					image.removeAttribute( 'srcset' );
-				}
-			}
+	// trickery to remove it from the dom / not load the image
+	// TODO: test if this is necessary
```
Comment (samouri) [inline client/lib/post-normalizer/rule-content-safe-images.js]: specifically I was talking about testing if its necessary to change src to transparent gif for fakedom to not pre-fetch images

File: client/lib/post-normalizer/rule-content-safe-images.js [hunk 3]
```diff
@@ -1,160 +1,86 @@
...
-
-			if ( ! safeSource || imageShouldBeRemovedFromContent( imgSource ) ) {
-				image.parentNode.removeChild( image );
-				// fun fact: removing the node from the DOM will not prevent it from loading. You actually have to
-				// change out the src to change what loads. The following is a 1x1 transparent gif as a data URL
-				image.setAttribute( 'src', TRANSPARENT_GIF );
-				image.removeAttribute( 'srcset' );
-				return;
-			}
+/** Checks whether or not imageUrl shoudl be removed from the dom
```
Comment (blowery) [inline client/lib/post-normalizer/rule-content-safe-images.js]: speling on "should"

**Comment by samouri** [comment]
I'm going to split this into two more PRs.  PRs that include renaming in addition to other things are too difficult to make meaningful.

1 - just renaming
2 - rejiggering the codes

File: client/state/reader/posts/normalization-rules.js [hunk 4]
```diff
@@ -141,15 +141,14 @@ const fastPostNormalizationRules = flow( [
 	firstPassCanonicalImage,
 	withContentDom( [
 		removeStyles,
-		makeImagesSafe( READER_CONTENT_WIDTH ),
-		discoverFullBleedImages,
 		makeEmbedsSafe,
+		partial( makeImagesSafe, partial.placeholder, partial.placeholder, READER_CONTENT_WIDTH ),
```
Comment (bluefuton) [inline client/state/reader/posts/normalization-rules.js]: Worth adding a comment explaining what this does I think.

**Comment by bluefuton** [comment]
Shall we break out the `client/lib/safe-image-url/index.js` cleanup into a separate small PR?

**Comment by samouri** [comment]
@bluefuton you are definitely right! I'll split it into two more. 

This PR is a hydra

**Pull request closed without merging**